yt_dlp/extractor/ilpost.py

```diff
@@ -35,34 +37,75 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, display_id)
 
         endpoint_metadata = self._search_json(
-            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', display_id)
-        episode_id = endpoint_metadata['post_id']
-        podcast_id = endpoint_metadata['podcast_id']
-        podcast_metadata = self._download_json(
-            endpoint_metadata['ajax_url'], display_id, data=urlencode_postdata({
-                'action': 'checkpodcast',
-                'cookie': endpoint_metadata['cookie'],
-                'post_id': episode_id,
-                'podcast_id': podcast_id,
-            }))
+            r'{"props":{"pageProps":', webpage, 'metadata', display_id)
+        episode_id = endpoint_metadata['data']['data']['episode']['data'][0]['id']
+        podcast_id = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0, 'parent', 'id'))
+        podcast_metadata = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0))
 
-        episode = traverse_obj(podcast_metadata, (
-            'data', 'postcastList', lambda _, v: str(v['id']) == episode_id, {dict}), get_all=False)
+        episode = podcast_metadata
```

Comment:
you don't always need to make new code fit into the old code's logic; feel free to change var names instead of duplicating them, etc